docs/source/en/hf_quantizer.md

```diff
@@ -0,0 +1,67 @@
+```py
+class Linear4bit(nn.Module):
+    def __init__(self, ...):
+        ...
+    
+    def forward(self, x):
+        return my_4bit_kernel(x, self.weight, self.bias)
+```
+That way, transformers models can be easily quantizable by simply replacing some instances of `nn.Linear` with a target class.
+- Ideally the quantization method should be serializable, i.e. you can save the quantized weights locally or push them on the Hub.
+- Make sure the package that contains the quantization kernels / primitive is mature enough (e.g. no frequent breaking changes).
+
+Note that for some quantization methods it is a strong requirement to "pre-quantize" the models through data calibration (e.g. AWQ). In that case, we prefer to support only inference through transformers and let third-party libraries maintained by the ML community deal with the model quantization itself.
+
+## How should I get started?
+
+0- 📕 Create a new quantization config class inside `src/transformers/utils/quantization_config.py`, and make sure to expose that new quantization config inside transformers main init, by adding it on the `_import_structure` object of `src/transformers/__init__.py`.
+
+1-  🗃 Create a new file inside `src/transformers/quantizers/` named `quantizer_your_method.py` and make it inherit from `src/transformers/quantizers/base.py::HFQuantizer`. Make sure to add the new quantizer and quantization config in the quantization auto-mapping in `src/transformers/quantizers/auto.py`
+
+2- 🔩 Define the following class attributes / property methods:
+
+2.1. `requires_calibration`: Whether the quantization method requires a data-calibration process. If set to `True` you'll be able to only support inference (with quantized weights) and not inference + quantization.
+2.2. `required_packages`: A list of strings of the required packages to use the quantized weights. You might need to define some new utility methods such as `is_auto_awq_available` in `transformers/src/utils/import_utils.py`
+2.3 `requires_parameters_quantization`: (Advanced - defaults to `False`) Only required if your quantization methods requires a special care of the underlying `nn.Parameter` object. For example bitsandbytes uses `Params4bit` and `Int8Param` that requires some special care when quantizing the model. Most of the recent quantization method packs int2 / int4 weights inside `torch.uint8` weights so that flag should not be really required
+2.4 `is_serializable`: A property method to determine whether the method is serializable or not
+2.5. `is_trainable`:  A property method to determine whether you can fine-tune models on top of that quantization methods (with or without PEFT approaches).
+
+
+3- 🪛 Write the `validate_environment` and `set_torch_dtype` methods. These methods are called before creating the quantized model to make sure users are on the right configuration. You can have a look at how this is done on other quantizers.
```

Comment:
Maybe add a link to where users can look at an example of this for other quantizers?

```suggestion
3- 🪛 Write the `validate_environment` and `set_torch_dtype` methods. These methods are called before creating the quantized model to ensure users use the right configuration. You can have a look at how this is done on other quantizers.
```